From 36925167f4aec9e9f94ceee6ce9fa96d745a6df0 Mon Sep 17 00:00:00 2001
From: Czarek Nakamoto <cyjan@mrcyjanek.net>
Date: Tue, 2 Apr 2024 16:51:56 +0200
Subject: [PATCH 01/17] fix missing ___clear_cache when targetting iOS

Point external/randomx at MAGICGrants/RandomX fix-ios: RandomX v1.2.3, the
version monero pins, plus the iOS fix from MrCyjaneK/RandomX cyjan-fix-ios
(sys_icache_invalidate instead of __builtin___clear_cache on Apple), with
the original icache flush ranges kept.
---
 .gitmodules      | 3 ++-
 external/randomx | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/.gitmodules b/.gitmodules
index 31c6921e5..0b92502c2 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -6,7 +6,8 @@
 	url = https://github.com/trezor/trezor-common.git
 [submodule "external/randomx"]
 	path = external/randomx
-	url = https://github.com/tevador/RandomX
+	url = https://github.com/MAGICGrants/RandomX
+	branch = fix-ios
 [submodule "external/supercop"]
 	path = external/supercop
 	url = https://github.com/monero-project/supercop
diff --git a/external/randomx b/external/randomx
index 12f2c2ffe..9d97dc51d 160000
--- a/external/randomx
+++ b/external/randomx
@@ -1 +1 @@
-Subproject commit 12f2c2ffe2108d6cf54c391fee33c8bc3646cdab
+Subproject commit 9d97dc51d1097055bea68145bb139dfee278c831
-- 
2.50.1 (Apple Git-155)

